### PR TITLE
Hal1.0

### PR DESCRIPTION
**Pull request by emilwestergren**
* Fixed broken HAL 1.0 code
* Moved general code all the way up to the base class `CameraWrapper`, this way I could remove some classes

@sebastianbaginski peer review

### CODE REVIEW

File: HAL/CameraWrapper.h [hunk 1]
```diff
@@ -57,19 +60,32 @@ typedef struct {
...
-	CameraWrapper() {}
+	CameraWrapper();
 	virtual ~CameraWrapper() {}
 	sp<GraphicBuffer> getCameraBuffer(int width, int height, int format, int usage);
 	void recycleCameraBuffer(sp<GraphicBuffer> buffer);
 	void freeCameraBuffers();
 	virtual int openCameraDevice(struct hw_device_t **hw_device, camera_module_t *wrappedModule) = 0;
 	static create_method_t createMethod;
-	virtual stride_t getStride(const sp<GraphicBuffer>& buffer) = 0;
+	virtual stride_t getStride(const sp<GraphicBuffer>& buffer) { stride_t result; result.horizontal = buffer->width; result.vertical = buffer->height; return result; }
```
Comment (sebastianbaginski) [inline HAL/CameraWrapper.h]: If you give `stride_t` a constructor, you could simplify this
```C++
virtual stride_t getStride(const sp<GraphicBuffer>& buffer) { return stride_t(buffer->width, buffer->height); }
```
But this is quite large PR already, your call.

File: HAL/HAL1/VideoProcessor.cpp [hunk 2]
```diff
@@ -45,7 +39,11 @@ void VideoProcessor::timestampCallback(int64_t timestamp, int32_t msg_type, cons
 	status_t result = inputBuffer->lock(GraphicBuffer::USAGE_SW_WRITE_OFTEN, &inputPtr);
 	memcpy(inputPtr, base, size);
 	inputBuffer->unlock();
-	this->processVideoCapture(inputBuffer, outputBuffer);
+	frame_data_t frame;
+	frame.focalLength = this->focalLength;
+	frame.inputBuffer = inputBuffer;
+	frame.outputBuffer = outputBuffer;
```
Comment (sebastianbaginski) [inline HAL/HAL1/VideoProcessor.cpp]: Same here, `frame_data_t` could have a constructor for simplicity.

File: nexus6p/VidhanceProcessor.h [hunk 3]
```diff
@@ -6,22 +6,21 @@
...
 #define VidhanceProcessor_H_
-#include "../HAL/AbstractVidhanceProcessor.h"
-#include "../HAL/SensorLogger.h"
+#include "../HAL/HAL3/DoubleBufferVideoProcessor.h"
 namespace imint {
 
 /* Predefined aligned width in bytes for RGBA8888 GraphicBuffers on Nexus 6p */
 #define ALIGNED_WIDTH_COUNT 10
-const aligned_width alignedWidth = { { 128, 256, 384, 512, 1024, 1536, 2560, 3072, 3584, 4608 }, ALIGNED_WIDTH_COUNT};
+const aligned_width _alignedWidth = { { 128, 256, 384, 512, 1024, 1536, 2560, 3072, 3584, 4608 }, ALIGNED_WIDTH_COUNT};
```
Comment (sebastianbaginski) [inline nexus6p/VidhanceProcessor.h]: If it is used only in this file it could be `static const`.

**Comment by sebastianbaginski** [comment]
OK.

File: nexus6p/VidhanceProcessor.h (hunk 3 above)
Comment (emilwestergren) [inline nexus6p/VidhanceProcessor.h]: Would `static` make any difference?

File: HAL/CameraWrapper.h (hunk 1 above)
Comment (emilwestergren) [inline HAL/CameraWrapper.h]: True

File: nexus6p/VidhanceProcessor.h (hunk 3 above)
Comment (sebastianbaginski) [inline nexus6p/VidhanceProcessor.h]: Symbols which are `static` at the file scope won't get exported to other compilation units, which prevents any possible name collisions. Also matter of style, when you see something `static` at the file scope, then you can be sure that it is not referenced in other units (and possibly changed).

**Pull request merged**